projects/ngneat/helipopper/src/lib/tippy.directive.ts

```diff
@@ -275,6 +277,38 @@ export class TippyDirective implements OnChanges, AfterViewInit, OnDestroy, OnIn
       });
   }
 
+  private handleEscapeButton() {
+    this.pressButton$(document.body, 'Escape')
```

Comment:
We are fine with only listening to `escape` and close the modal. You also don't want to listen if the input is `false`.

Reply:
We can run it outside the zone to save the appRef tick().

Reply:
> We are fine with only listening to escape and close the modal. 

You mean, that we don't need listening `hover$` & `focus$`? If yes,  the focus$ method can still be useful to fix another issue. Because when changing an item with the keyboard, we can still open the rest of the poppers. 

![popper issue](https://user-images.githubusercontent.com/84960503/120281566-62f69e80-c2b9-11eb-8712-bd051664380f.png)


> You also don't want to listen if the input is `false`.

So, in that case we should init this function like that: `this.hideOnEscapeButton && this.handleEscapeButton();`?
Then I could remove this `filter(() => this.hideOnEscapeButton)`

Reply:
I'm not sure I understand how the focus is related to escaping

Reply:
I am just wondering whether just starting another popper, or even leaving the button with an active popper, should not close it automatically. But maybe let's get back to the topic.

Can you tell me if such a solution would already be appropriate:
`this.hideOnEscapeButton && this.handleEscapeButton();`

```
  private handleEscapeButton() {
    this.zone.run(() => {
      this.pressButton$(document.body, 'Escape')
        .pipe(
          filter(() => this.instance.state.isVisible),
          takeUntil(this.destroyed)
        )
        .subscribe(() => this.hide());
    });
  }
```

Reply:
I am just wondering whether just starting another popper, or even leaving the button with an active popper, should not close it automatically. But maybe let's get back to the topic.

Can you tell me if such a solution would already be appropriate:
`this.hideOnEscapeButton && this.handleEscapeButton();`

```
  private handleEscapeButton() {
    this.zone.run(() => {
      this.pressButton$(document.body, 'Escape')
        .pipe(
          filter(() => this.instance.state.isVisible),
          takeUntil(this.destroyed)
        )
        .subscribe(() => this.hide());
    });
  }
```

Reply:
I'd only add the subscription when you open the popper and unsubscribe upon close